### ISSUE TITLE
Improve Implicit Resource Creation

### ISSUE THREAD

**Issue by spjmurray** [eschercloudai/unikorn#308]
The way implicit creation is implemented is the simplest possible way, e.g. referencing a project or control plane will create one.  Problem is this leads to some interesting side effects, for example listing clusters in a non-existent control plane will create that control plane.  One instance of things behaving weird in this regard is when the project is rescoped.  In the world of reactive javascript it's quite easy to trigger a list of clusters in a control plane that was valid for the old project, before the control plane list is updated, and things magically spring into existence.

Long story short, we should only create resources implicitly when the method is a POST to prevent bloat at the UX layer.
